app/src/location/location.service.ts

```diff
@@ -0,0 +1,26 @@
+import {Injectable} from '@nestjs/common';
+import * as MapboxGeocoding from '@mapbox/mapbox-sdk/services/geocoding';
+
+@Injectable()
+export class LocationService {
```

Comment:
If there was a requirement for the application to support multiple Geolocation providers and be easily configurable to swap them out, how would you structure your application?

Reply:
In this case, we need to add an extra layer "GeoLocation" which can use a lot of different providers inside and can be configured by env or database.